File: Easy/746. Min Cost Climbing Stairs/746. Min Cost Climbing Stairs.ts
function minCostClimbingStairs(cost: number[]): number {
    const n = cost.length;
    if (n === 2) {
        return Math.min(cost[0], cost[1]);
    }

    let prev1 = cost[0];
    let prev2 = cost[1];

    for (let i = 2; i < n; i++) {
        const current = cost[i] + Math.min(prev1, prev2);
        prev1 = prev2;
        prev2 = current;
    }

    return Math.min(prev1, prev2);
}

File: Easy/746. Min Cost Climbing Stairs/746. Min Cost Climbing Stairs.rs
impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let n = cost.len();
        if n == 2 {
            return cost[0].min(cost[1]);
        }

        let mut prev1 = cost[0];
        let mut prev2 = cost[1];

        for i in 2..n {
            let current = cost[i] + prev1.min(prev2);
            prev1 = prev2;
            prev2 = current;
        }

        prev1.min(prev2)
    }
}
